linkedList.js

```diff
@@ -3,6 +3,153 @@
  * @description  Main Module
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
-function linkedListGenerator(){
+function linkedListGenerator() {
+  let head = null;
+  let tail = null;
 
+  function getHead() {
+    return head;
+
+  }
+
+  function getTail() {
+    return tail;
+  }
+
+  function addNode(val) {
+    let node = {
+      value: val,
+      next: null
+    };
+
+    if (!head && !tail) {
+      head = node;
+      tail = node;
+    } else {
+      let currentTail = tail;
+      currentTail.next = node;
+      tail = node;
+    }
+
+    return node;
+  }
+
+  function get(number) {
+    let currentNode = head;
+    let count = 0;
+
+    while (currentNode.next) {
+
+      if (count === number) {
+        return currentNode;
+      } else {
+        count++;
+        currentNode = currentNode.next;
+      }
+
+    }
+
+    if (count === number) {
+      return tail;
+    } else {
+      return false;
+    }
+
+  }
+
+  function remove(number) {
+    let currentNode = head;
+    let previousNode = null;
+    let nextNode = null;
+    let count = 0;
+
+    while (currentNode) {
+
+      if (count === number) {
+
+        if (!nextNode && !previousNode) {
+
+          if (currentNode.next) {
```

Comment:
This seems counter intuitive since the if statement above is hinting that there's not next or previous. Maybe it's best to default nextNode to something like `nextNode = currentNode.next`?